<launch>
    <node pkg="yolov7-ros" type="detect_ros.py" name="detect" output="screen"  ns="yolov7">
        <!-- Download the official weights from the original repo -->
        <!-- 7 OK, yolov7-w6 OK, 7x too slow,e6e too slow, -->
        <param name="weights_path" type="str" value="$(find yolov7-ros)/weights/yolov7-w6.pt" />
        <!-- topic name to subscribe to -->
        <param name="img_topic" type="str" value="/zed2i/zed_node/rgb/image_rect_color" />
        <!-- topic name for the detection output -->
        <param name="out_topic" type="str" value="yolov7" />
        <!-- confidence threshold -->
        <param name="conf_thresh" type="double" value="0.5" />
        <!-- intersection over union threshold -->
        <param name="iou_thresh" type="double" value="0.6" />
        <!-- queue size for publishing -->
        <param name="queue_size" type="int" value="1" />
        <!-- image size to which to resize each input image before feeding into the
        network (the final output is rescaled to the original image size)   This cna be 640 or 1280 -->
        <param name="img_size" type="int" value="640" />
        <!-- flag whether to also publish image with the visualized detections -->
        <param name="visualize" type="bool" value="true" />
        <!-- 'cuda' or 'cpu' -->
        <param name="device" type="str" value="cuda" />
        <!-- for now only coco names -->
        <param name="yaml" type="str" value="$(find yolov7-ros)/conf/coco.yaml" />
    </node>
</launch>
